src/roulette.rs

```diff
@@ -25,44 +25,50 @@ pub struct PizzaDetail {
 }
 
 #[derive(Debug, Clone, Deserialize)]
-pub struct RouletteMessage {
-    phrase: String,
+pub struct FortunePhraseEntry {
+    pub phrase: String,
 }
 
-impl Deserializable for PizzaDetail {}
-impl Deserializable for RouletteMessage {}
+impl Deserializable for PizzaEntry {}
+impl Deserializable for FortunePhraseEntry {}
 
 lazy_static! {
-    static ref PIZZA_OPTIONS: Vec<PizzaDetail> = load_from_json(JsonFile::Pizza);
-    static ref ROULETTE_MESSAGES: Vec<RouletteMessage> = load_from_json(JsonFile::Roulette);
+    static ref PIZZAS: Vec<PizzaEntry> = load_from_json(JsonFile::Pizza);
+    static ref FORTUNE_PHRASES: Vec<FortunePhraseEntry> = load_from_json(JsonFile::FortunePhrase);
 }
 
 fn load_from_json<T: Deserializable>(file: JsonFile) -> Vec<T> {
     let json = match file {
-        JsonFile::Pizza => include_str!("../config/pizza.json"),
-        JsonFile::Roulette => include_str!("../config/roulette.json"),
+        JsonFile::FortunePhrase => include_str!("../config/fortune_phrases.json"),
+        JsonFile::Pizza => include_str!("../config/pizzas.json"),
     };
-    let result: Vec<T> = serde_json::from_str(json).expect("Failed to parse JSON configuration");
-    result
+
+    serde_json::from_str(json).expect("Failed to parse JSON configuration")
 }
 
-pub fn get_random_pizza(mode: SpinMode) -> PizzaDetail {
+fn get_random_element<T>(source: Vec<T>) -> Option<T>
```

Comment:
I think this does an unnecessary clone? I think we can return Option<&T

Reply:
Thanks for great feedback! If this function does not clone the result, you will run into issues since the function will return a reference to its function parameter.

Reply:
I thiiink I found a way!